File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/configuration/JacksonConfiguration.java
package com.jmsoftware.maf.springcloudstarter.configuration;

import cn.hutool.core.util.StrUtil;
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer;
import com.jmsoftware.maf.common.constant.UniversalDateTime;
import lombok.RequiredArgsConstructor;
import lombok.val;
import org.springframework.boot.autoconfigure.jackson.Jackson2ObjectMapperBuilderCustomizer;
import org.springframework.boot.autoconfigure.jackson.JacksonProperties;
import org.springframework.context.annotation.Bean;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

/**
 * Description: LocalDateTimeSerializerConfiguration, change description here.
 *
 * @author Johnny Miller (锺俊), email: johnnysviva@outlook.com, date: 6/27/2021 3:57 PM
 **/
@RequiredArgsConstructor
public class JacksonConfiguration {
    private final JacksonProperties jacksonProperties;

    @Bean
    public LocalDateTimeSerializer localDateTimeSerializer() {
        var pattern = UniversalDateTime.DATE_TIME_FORMAT;
        if (StrUtil.isNotBlank(this.jacksonProperties.getDateFormat())) {
            pattern = this.jacksonProperties.getDateFormat();
        }
        return new LocalDateTimeSerializer(DateTimeFormatter.ofPattern(pattern));
    }

    @Bean
    public Jackson2ObjectMapperBuilderCustomizer jackson2ObjectMapperBuilderCustomizer(LocalDateTimeSerializer localDateTimeSerializer) {
        var pattern = UniversalDateTime.DATE_TIME_FORMAT;
        if (StrUtil.isNotBlank(this.jacksonProperties.getDateFormat())) {
            pattern = this.jacksonProperties.getDateFormat();
        }
        val localDateTimeDeserializer = new LocalDateTimeDeserializer(DateTimeFormatter.ofPattern(pattern));
        return jackson2ObjectMapperBuilder -> jackson2ObjectMapperBuilder
                .serializerByType(LocalDateTime.class, localDateTimeSerializer)
                .deserializerByType(LocalDateTime.class, localDateTimeDeserializer)
                .serializerByType(Long.class, ToStringSerializer.instance);
    }
}

File: spring-cloud-starter/src/main/kotlin/com/jmsoftware/maf/springcloudstarter/configuration/JacksonConfiguration.kt
package com.jmsoftware.maf.springcloudstarter.configuration

import com.fasterxml.jackson.databind.ser.std.ToStringSerializer
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer
import com.fasterxml.jackson.datatype.jsr310.ser.OffsetDateTimeSerializer
import com.fasterxml.jackson.datatype.jsr310.ser.ZonedDateTimeSerializer
import com.jmsoftware.maf.common.util.logger
import org.springframework.boot.autoconfigure.jackson.Jackson2ObjectMapperBuilderCustomizer
import org.springframework.context.annotation.Bean
import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder
import java.time.LocalDateTime
import java.time.OffsetDateTime
import java.time.ZonedDateTime

/**
 * # JacksonConfiguration
 *
 * Description: JacksonConfiguration, change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/14/22 10:33 AM
 */
class JacksonConfiguration {
    companion object {
        private val log = logger()
    }

    @Bean
    fun jackson2ObjectMapperBuilderCustomizer(): Jackson2ObjectMapperBuilderCustomizer {
        return Jackson2ObjectMapperBuilderCustomizer { jackson2ObjectMapperBuilder: Jackson2ObjectMapperBuilder ->
            jackson2ObjectMapperBuilder
                // Serialization & deserialization of `LocalDateTime`
                .serializerByType(LocalDateTime::class.java, LocalDateTimeSerializer.INSTANCE)
                .deserializerByType(LocalDateTime::class.java, LocalDateTimeDeserializer.INSTANCE)
                // Serialization of `ZonedDateTime`
                .serializerByType(ZonedDateTime::class.java, ZonedDateTimeSerializer.INSTANCE)
                // Serialization of `OffsetDateTime`
                .serializerByType(OffsetDateTime::class.java, OffsetDateTimeSerializer.INSTANCE)
                // Serialization of `Long`
                .serializerByType(Long::class.java, ToStringSerializer.instance)
        }.also {
            log.info("Customized Jackson ObjectMapper")
        }
    }
}
